### ISSUE TITLE
Interpolate service

### ISSUE THREAD

**Issue by ocombe** [angular/angular#4889]
Hello,

I'm writing a service (for [ng2-translate](https://github.com/ocombe/ng2-translate)) where I need to parse strings containing interpolated values (something like: `{{greeting}} {{name}}!`). I cannot use a component to do that, that would defeat the purpose of what I am doing.
In Angular 1 there was this very useful `$interpolate` service: https://docs.angularjs.org/api/ng/service/$interpolate

I've searched the Angular 2 source code and the only thing that I've found is the `Parser` class that seems to be used by the templates, but I haven't been able to use it correctly to do what I wanted and in the end I haven't been able to find anything even remotely similar to the `$interpolate` service from Angular 1. Is there any way to parse a string like that without using a component and a template?
If not, could this be something that you would consider adding? It's probably something that will be needed for i18n as well.

**Comment by pkozlowski-opensource** [comment]
There is no direct equivalent on `$interpolate` in ng2. The closest thing would be: https://github.com/angular/angular/blob/860e88c5becd53221e9f54564e430a9df794c0ab/modules/angular2/src/core/change_detection/parser/parser.ts#L94 but it pretty low-level. 

I guess you would like to use it in your service, right? I guess the best solution is to hand-roll sth for now. There are no plans for now to have direct equivalent of $interpolate.

**Comment by ocombe** [comment]
Yeah I tried to use parseInterpolation but couldn't make it work (there is no documentation/comments for AST and after trying for 3h to understand the thing, I gave up). Whenever you have some free time I'd appreciate your help to make this work because it looks like a good idea for a new ng2 library :)
Until then, I wrote my own naive parser that only works with objects: https://github.com/ocombe/ng2-translate/blob/master/src/translate.parser.ts

Still, I think that this would be an interesting addition to angular 2 and I think that @robwormald agreed with me on that. I understand that it is probably not a blocking feature request for the beta, but maybe for later.
